source/_docs/guides/jenkins.md

```diff
@@ -0,0 +1,223 @@
+
+2. Give the project a name. Do not use spaces. Select "Freestyle Project" and "OK" to save.
+
+![New Jenkins project](/source/docs/assets/images/integrations/jenkins/new_job.png)
+
+3. In the General section, check the box to select "GitHub project" and enter the repo url (e.g. https://github.com/YOUR-ORG/YOUR-PROJECT).
+
+4. In the Source Code Management section, add:
+- Repository URL: The path to the .git file of your repository, e.g. https://github.com/YOUR-ORG/YOUR-PROJECT.git 
+- Name: origin
+- Refspec: +refs/heads/*:refs/remotes/origin/*
+- Branches specifier: origin/*
+- Additional Behaviours: Prune stale remote-tracking branches, and Clean after checkout
+
+![SCM view](/source/docs/assets/images/integrations/jenkins/scm_settings.png)
+
+5. We want code changes to trigger our build, as opposed to setting up a periodic build, for example. Check the box labelled, "GitHub hook trigger for GITScm polling."
+
+6. For "Build Environment," check the box labelled, "Inject environment variables to the build process." Verify the file path exists on the Jenkins server, or create a file and add the path accordingly. Add these variables to the "Properties Content" field, one per line, with no quotations marks.
+
+- TERMINUS_TOKEN=enter an existing token or [create a new token](https://pantheon.io/docs/machine-tokens) from your user dashboard for Jenkins.
+- SITE_ID=<your site name.
+
+![Env vars view](/source/docs/assets/images/integrations/jenkins/env_vars.png)
+  
+## Step 5: Add Build Steps
+These tasks will execute in sequence, and the job will quit if any fail. Add these in separate build steps, selecting "Execute shell" for all steps.
+
+Jenkins logs into Pantheon:
+```nohighlight
+echo "Logging into Terminus"
+terminus auth:login --machine-token=${TERMINUS_TOKEN}
+```
+Verifies the dev site is awake and in git mode.
+```nohighlight
+echo "Waking Dev environment."
+terminus env:wake -n ${SITE_ID}.devterminus env:wake -n $SITE_ID.dev
+```
+
+```nohighlight
+echo "Setting site to git mode."
+terminus connection:set ${SITE_ID}.dev git
+```
+Jenkins creates a multidev and pushes the new code to this environment
+```nohighlight
+echo "Creating multidev"
+cd ${WORKSPACE}
+terminus build-env:create ${SITE_ID}.dev ci-${BUILD_ID} --yes
+```
+
+```nohighlight
+echo "Run database updates and clear cache"
+terminus drush -n ${SITE_ID}.ci-${BUILD_ID} -- updatedb -y
+terminus drush ${SITE_ID}.ci-${BUILD_ID} cr
+```
+Then the test suite we include with the example is run.
+```nohighlight
+echo "Running behat"
+TERMINUS_ENV=ci-$BUILD_ID TERMINUS_SITE=$SITE_ID $WORKSPACE/tests/scripts/run-behat
+```
+
+Add one Conditional (single step) build task, to merge the code from the Pantheon multidev to the pantheon/master, i.e. your dev site on Pantheon. This will only happen when changing the master branch.
+
+Conditions:
+```nohighlight
+Run: Regular Expression Match
+Expression: (?i).*origin/master.*
+Label: ${ENV,var="GIT_BRANCH"}
```

Comment:
If we are using conditionals here, then maybe we don't need to clone the project to do the merge-to-dev step?